addons/csw/functions/fnc_ai_reload.sqf

```diff
@@ -6,91 +6,68 @@
-                if (_loadInfo select 0) then {
-                    _reloadMag = _x;
-                    _reloadSource = _xSource;
-                    _reloadNeededAmmo = _loadInfo select 2;
-                    TRACE_3("found mag",_reloadMag,_reloadSource,_x);
-                    breakOut "findSource";
-                };
-            };
-        } forEach _cswMagazines;
-    } forEach _compatibleMags;
-} forEach _nearSupplies;
-if (_reloadMag == "") exitWith {TRACE_1("could not find mag",_reloadMag);};
+if (_clearForcedMag) then {
+    _vehicle setVariable [QGVAR(forcedMag), nil, true];
+};
 
-// Figure out what we can add from the magazines we have
-private _bestAmmoToSend = -1;
-{
-    _x params ["_xMag", "_xAmmo"];
-    TRACE_2("",_xMag,_xAmmo);
-    if (_xMag == _reloadMag) then {
-        if ((_bestAmmoToSend == -1) || {(_xAmmo > _bestAmmoToSend) && {_xAmmo <= _reloadNeededAmmo}}) then {
-            _bestAmmoToSend = _xAmmo;
-        };
-    };
-} forEach (if (_reloadSource isKindOf "CAManBase") then {magazinesAmmo _reloadSource} else {magazinesAmmoCargo _reloadSource});
-TRACE_4("",_reloadSource,_reloadMag,_reloadNeededAmmo,_bestAmmoToSend);
-if (_bestAmmoToSend == -1) exitWith {ERROR("No ammo");};
+if (_magazineInfo isEqualTo []) exitWith {};
+_magazineInfo params ["_carryMag", "_turretPath", "_loadInfo", "_magSource", "", "_ammo"];
 
 // Remove the mag from the source
-[_reloadSource, _reloadMag, _bestAmmoToSend] call EFUNC(common,removeSpecificMagazine);
+[_magSource, _carryMag, _ammo] call EFUNC(common,removeSpecificMagazine);
 
-private _timeToLoad = 1;
-if (!isNull(configOf _vehicle >> QUOTE(ADDON) >> "ammoLoadTime")) then {
-    _timeToLoad = getNumber(configOf _vehicle >> QUOTE(ADDON) >> "ammoLoadTime");
+// AI never returns ammo and removes the magazine before reloading, so we can skip distance and weaponHolder checks
+private _eventParams = [_vehicle, _turretPath, objNull, _carryMag, _ammo, _gunner];
+
+if (_instantReload) exitWith {
+    TRACE_1("calling addTurretMag event: instant AI reload",_this);
+    [QGVAR(addTurretMag), _eventParams] call CBA_fnc_globalEvent;
 };
 
+private _timeToLoad = GET_NUMBER(configOf _vehicle >> QUOTE(ADDON) >> "ammoLoadTime",1);
+
 TRACE_1("Reloading in progress",_timeToLoad);
 [{
-    params ["_vehicle", "_turretPath", "_gunner", "_reloadMag", "_bestAmmoToSend"];
-    if ((!alive _vehicle) || {!alive _gunner} || {(_vehicle distance _gunner) > 10}) exitWith {TRACE_1("invalid state",_this);};
+    params ["_vehicle", "", "", "", "", "_gunner"];
+    if !(alive _vehicle && {alive _gunner}) exitWith {TRACE_2("invalid state",alive _vehicle,alive _gunner);};
 
     // Reload the static weapon
-    TRACE_5("calling addTurretMag event",_vehicle,_turretPath,_gunner,_reloadMag,_bestAmmoToSend);
+    TRACE_1("calling addTurretMag event: AI reload",_this);
     [QGVAR(addTurretMag), _this] call CBA_fnc_globalEvent;
```

Comment:
This needs to only run where the turret is local. Why is it run globally? We have a CBA turret event.